libnavigator/src/main/java/com/mapbox/navigation/navigator/internal/MapboxNativeNavigatorImpl.kt

```diff
@@ -262,6 +278,7 @@ object MapboxNativeNavigatorImpl : MapboxNativeNavigator {
      *
      * @return `true` if route is different, `false` otherwise.
      */
+    // TODO make async after https://github.com/mapbox/mapbox-navigation-native/issues/4164
```

Comment:
@korshaknn As this PR is introducing breaking changes, should we prepare now for when this is available so we don't introduce SemVer later? Wondering if we can expose the callback right now and then when NN has the fix we can just use it i.e. introduce breaking changes now and don't worry later after 2.0 is out.

Reply:
@Guardiola31337 NN changes won't affect SDK side. Method signature will be the same.
`withContext` will be replaces with `suspendCancellableCoroutine` and that's it

Reply:
I was asking because wasn't sure if we were exposing this publicly. Similar to https://github.com/mapbox/mapbox-navigation-android/blob/778cea691ea5581b4713e9e78c1134a6ad27486c/libnavigation-core/api/current.txt#L22 for example. If that's not the case, even easier 💃

Reply:
@Guardiola31337 they are not the same. 
- `updateLegIndex` was a blocking direct call func -> NN changes to async -> need to change signature
- `isDifferentRoute` is `suspend` at the moment -> NN  changes to async -> no need to update signature, just replace 
`withContext`  with `suspendCancellableCoroutine`